botocore/data/endpoints.json

```diff
@@ -0,0 +1,1200 @@
+            "eu-west-1": {},
+            "eu-central-1": {}
+          }
+        },
+        "elasticfilesystem": {
+          "endpoints": {
+            "us-west-2": {}
+          }
+        },
+        "elasticloadbalancing": {
+          "defaults": {
+            "protocols": [
+              "http",
+              "https"
+            ]
+          },
+          "endpoints": {
+            "us-east-1": {},
+            "us-west-1": {},
+            "us-west-2": {},
+            "ap-northeast-1": {},
+            "ap-northeast-2": {},
+            "ap-southeast-1": {},
+            "ap-southeast-2": {},
+            "sa-east-1": {},
+            "eu-west-1": {},
+            "eu-central-1": {}
+          }
+        },
+        "elasticmapreduce": {
+          "defaults": {
+            "protocols": [
+              "http",
+              "https"
+            ],
+            "sslCommonName": "{region}.{service}.{dnsSuffix}"
+          },
+          "endpoints": {
+            "us-east-1": {
+              "sslCommonName": "{service}.{region}.{dnsSuffix}"
+            },
+            "us-west-1": {},
+            "us-west-2": {},
+            "ap-northeast-1": {},
+            "ap-northeast-2": {},
+            "ap-southeast-1": {},
+            "ap-southeast-2": {},
+            "sa-east-1": {},
+            "eu-west-1": {},
+            "eu-central-1": {
+              "sslCommonName": "{service}.{region}.{dnsSuffix}"
+            }
+          }
+        },
+        "elastictranscoder": {
+          "endpoints": {
+            "us-east-1": {},
+            "us-west-1": {},
+            "us-west-2": {},
+            "ap-northeast-1": {},
+            "ap-southeast-1": {},
+            "eu-west-1": {}
+          }
+        },
+        "email": {
+          "endpoints": {
+            "us-east-1": {},
+            "us-west-2": {},
+            "eu-west-1": {}
+          }
+        },
+        "glacier": {
+          "defaults": {
+            "protocols": [
+              "http",
+              "https"
+            ]
+          },
+          "endpoints": {
+            "us-east-1": {},
+            "us-west-1": {},
+            "us-west-2": {},
+            "ap-northeast-1": {},
+            "ap-northeast-2": {},
+            "ap-southeast-2": {},
+            "eu-west-1": {},
+            "eu-central-1": {}
+          }
+        },
+        "iam": {
+          "partitionEndpoint": "aws-global",
+          "isRegionalized": false,
+          "endpoints": {
+            "aws-global": {
+              "hostname": "iam.amazonaws.com",
+              "credentialScope": {
+                "region": "us-east-1"
+              }
+            }
+          }
+        },
+        "importexport": {
+          "partitionEndpoint": "aws-global",
+          "isRegionalized": false,
+          "endpoints": {
+            "aws-global": {
+              "hostname": "importexport.amazonaws.com",
+              "signatureVersions": [
+                "v2"
```

Comment:
Interesting, must have missed that in the spec.  Didn't realize we were phasing out signature version from the model.

If that's the case, I'd prefer to move the default value of "v4" from botocore code into the endpoints json file somehow.  Seems like you should be able to derive what signature version to use entirely from the endpoints file.